staking/handler.go

```diff
@@ -34,42 +34,37 @@ func NewHandler(k Keeper, contractCaller helper.IContractCaller) sdk.Handler {
 
 // HandleMsgValidatorJoin msg validator join
 func HandleMsgValidatorJoin(ctx sdk.Context, msg types.MsgValidatorJoin, k Keeper, contractCaller helper.IContractCaller) sdk.Result {
-	k.Logger(ctx).Info("Handing new validator join", "msg", msg)
+	k.Logger(ctx).Info("Handling new validator join", "msg", msg)
 
-	if confirmed := contractCaller.IsTxConfirmed(msg.TxHash.EthHash()); !confirmed {
+	// get main tx receipt
+	receipt, err := contractCaller.GetConfirmedTxReceipt(msg.TxHash.EthHash())
+	if err != nil || receipt == nil {
 		return hmCommon.ErrWaitForConfirmation(k.Codespace()).Result()
 	}
 
-	// fetch validator from mainchain
-	validator, err := contractCaller.GetValidatorInfo(msg.ID)
-	if err != nil {
-		k.Logger(ctx).Error(
-			"Unable to fetch validator from rootchain",
-			"error", err,
-		)
-		return hmCommon.ErrNoValidator(k.Codespace()).Result()
+	// TODO remove this block in next release and use msg.LogIndex -- start
+	logIndex, found := contractCaller.FindStakedEventLogIndex(receipt)
```

Comment:
Can we have a parent function which does both `FindAndDecodeEventLog`, just to remove clutter from handler.

Reply:
This will get removed in cs-2003 and only `DecodeValidatorJoinEvent` will be used. We cannot remove `DecodeValidatorJoinEvent` due to different struct types.